examples/gke/xfs-formatter-daemonset.yaml

```diff
@@ -0,0 +1,92 @@
+# Daemonset that converts ephemeral-storage to xfs
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: xfs-formatter
+spec:
+  selector:
+    matchLabels:
+      app: xfs-formatter
+  template:
+    metadata:
+      labels:
+        app: xfs-formatter
+    spec:
+      hostPID: true
+      nodeSelector:
+        scylla.scylladb.com/gke-ephemeral-storage-local-ssd: "true"
+      tolerations:
+      - key: role
+        operator: Equal
+        value: scylla-clusters
+        effect: NoSchedule
+      containers:
+      - name: xfs-formatter
+        image: docker.io/scylladb/scylla-operator:latest
+        securityContext:
+          privileged: true
+        command:
+        - "/bin/bash"
+        - "-c"
+        - |
+          set -euExo pipefail
+          shopt -s inherit_errexit
+
+          cp -r /usr/local/lib/scylla-operator/gke/xfs-formatter /host/var/lib/
+
+          if [[ ! -d /host/usr/local/lib/systemd ]]; then
+            mkdir /host/usr/local/lib/systemd
+          fi
+
+          if [[ ! -d /host/usr/local/lib/systemd/system ]]; then
+            mkdir /host/usr/local/lib/systemd/system
+          fi
+
+          cp -r /usr/local/lib/scylla-operator/gke/systemd/. /host/usr/local/lib/systemd/system
+
+          if [[ ! -d /host/usr/local/lib/systemd/system/kube-node-configuration.service.d ]]; then
+            mkdir /host/usr/local/lib/systemd/system/kube-node-configuration.service.d
+          fi
+
+          cat<<-EOF > /host/usr/local/lib/systemd/system/kube-node-configuration.service.d/10-path.conf
+            [Service]
+            Environment="PATH=/var/lib/xfs-formatter/bin:$( chroot /host /bin/bash -c 'echo "${PATH}"' )"
+          EOF
+
+          if [[ ! -d /host/mnt/raid-disks ]]; then
+            mkdir /host/mnt/raid-disks
+          fi
+
+          if [[ ! -d /host/mnt/raid-disks/disk0 ]]; then
+            mkdir /host/mnt/raid-disks/disk0
+          fi
+          chmod a+w /host/mnt/raid-disks/disk0
```

Comment:
why `a+w`?

Reply:
I wanted to recreate what GKE was doing in [configure-helper.sh](https://github.com/kubernetes/kubernetes/blob/c2a633e07ec6129398dfef95c9e011ea5535ded2/cluster/gce/gci/configure-helper.sh#L283), which we're interfering with. What would you suggest instead?

Reply:
it's fine then